### DISCUSSION TITLE
Vite SSR Federation

### DISCUSSION THREAD

**Discussion by reiyanyan** [vikejs/vike#1162] [Discussion] [answered]
**Vue**

anyone had to tried [vite-plugin-federation](https://github.com/originjs/vite-plugin-federation)

`federation({
     ...
      shared: ["vue"]
})`

the build command of `vite build` produced an error below. But, ff`shared: ['vue']` was removed, the build was succeed, but the reactive of vue doesn't work :/

`Entry module "vue" cannot be external.`

**Comment by brillout** [comment]
> been stuck in there @brillout , any advice maybe?

From the information you provided, it seems that your issues are unrelated to Vike.

What's your use case for module federation?

**Reply by reiyanyan** [reply]
yes for module federation with ssr

**Reply by brillout** [reply]
I understand that. My question is: why do you want to use module federation?

**Reply by reiyanyan** [reply]
thats the client wanted for micro frontend

**Reply by brillout** [reply]
That isn't convincing: the more I know about your motivation the better.

**Reply by reiyanyan** [reply]
the Tech Lead of clients wanted to be micro frontend (seems like micro services backend behaviour) and they also want better SEO (i think SSR is the way)

**Comment by beneschjozsefzoltan** [comment]
This is my problem: https://github.com/vikejs/vike/discussions/1186

My use case is that I would like to reuse components from another frontend, where the exposes of componenets are already solved. 
I would like to generate static site where I would like to use the exposed components. 
I hope I can solve this problem with vike but maybe not. 
In dev mode my solution is working, but under the build process I got this error and i can't find a solution: [vite]: Rollup failed to resolve import "__federation_fn_satisfy" from "virtual:__federation_fn_import".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`

 @reiyanyan have you got any idea how can I solve this problem?

**Reply by reiyanyan** [reply, marked as answer]
the vite config below solved my problem for `__federation_fn_satisfy`

```
build {
   ...,
   rollupOptions: { external: ["__federation_fn_satisfy", "vue"] }
}
```

**Reply by beneschjozsefzoltan** [reply]
I tried. If I put it in the external this is the error, what i get:
```
Error: Cannot find package '__federation_fn_satisfy' imported from /Develop/repo/vike-ovbportal-frontend-ssg/dist-ssg/server/chunks/chunk-43dd7b58.js
    at new NodeError (node:internal/errors:405:5)
    at packageResolve (node:internal/modules/esm/resolve:782:9)
    at moduleResolve (node:internal/modules/esm/resolve:831:20)
    at defaultResolve (node:internal/modules/esm/resolve:1036:11)
    at DefaultModuleLoader.resolve (node:internal/modules/esm/loader:251:12)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:140:32)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:33)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

**Reply by reiyanyan** [reply]
try the top level build solution
[build-target](https://github.com/originjs/vite-plugin-federation#faq)
as my config

`build { ..., minify: true, target: [....] }`

**Reply by beneschjozsefzoltan** [reply]
Sry, but I'm not exactly understand how :(
